### PR TITLE
refactor teams.Storage, and plug into  FTL

### PR DESCRIPTION
**Pull request by maxtaco**

### CODE REVIEW

File: go/teams/storage.go [hunk 1]
```diff
@@ -66,170 +125,182 @@ func (s *Storage) Get(ctx context.Context, teamID keybase1.TeamID, public bool)
...
 // Store some TeamSigChainState's in memory. Threadsafe.
-type MemoryStorage struct {
+type memoryStorageGeneric struct {
 	libkb.Contextified
 	lru *lru.Cache
 }
 
-func NewMemoryStorage(g *libkb.GlobalContext) *MemoryStorage {
-	nlru, err := lru.New(MemCacheLRUSize)
+func newMemoryStorageGeneric(g *libkb.GlobalContext, sz int) *memoryStorageGeneric {
```
Comment (mlsteele) [inline go/teams/storage.go]: sz size

File: go/teams/storage.go [hunk 2]
```diff
@@ -239,7 +310,7 @@ func (s *MemoryStorage) key(teamID keybase1.TeamID, public bool) (key string) {
 
 // --------------------------------------------------
 
-func getLocalStorageSecretBoxKey(ctx context.Context, g *libkb.GlobalContext) (fkey [32]byte, err error) {
+func getLocalStorageSecretBoxKeyGeneric(ctx context.Context, g *libkb.GlobalContext, reason libkb.EncryptionReason) (fkey [32]byte, err error) {
```
Comment (mlsteele) [inline go/teams/storage.go]: this one doesn't really need generic in its name

File: go/teams/storage.go [hunk 3]
```diff
@@ -16,47 +16,106 @@ import (
...
+func (d *DiskStorageItem) setValue(v teamDataGeneric) error {
+	typed, ok := v.(*keybase1.TeamData)
+	if !ok {
+		return fmt.Errorf("teams.Storage#Put: Bad object for setValue; got type %T", v)
+	}
+	d.State = typed
+	return nil
+}
+
+func NewStorage(g *libkb.GlobalContext) *Storage {
```
Comment (mlsteele) [inline go/teams/storage.go]: add a comment acknowledging past mistakes so the next person doesn't spend time investigating
// DBChatInbox is used here. Oops. Next time the teams cache is busted we can switch it.

**Comment by maxtaco** [comment]
You know, I did that at first, but then split it up into 2 to make it easier to review.

**Comment by mlsteele** [comment]
Ok cool, can do it in a subsequent pr

**Comment by maxtaco** [comment]
yeah i can do that in -3

**Pull request merged**